src/pybind/mgr/ansible_orchestrator/module.py

```diff
@@ -0,0 +1,364 @@
+                                                 playbook = GET_INVENTORY_PLAYBOOK,
+                                                 logger = self.log,
+                                                 result_pattern = "RESULTS",
+                                                 params = "{}")
+
+        # Assing the process_output function
+        ansible_operation.process_output = process_inventary_json
+        ansible_operation.event_filter = "runner_on_ok"
+
+        # Execute the playbook to obtain data
+        ansible_operation.execute_playbook()
+
+        self.all_completions.append(ansible_operation)
+
+        return ansible_operation
+
+    def verify_config(self):
+
+        if not self.get_config('server_addr', ''):
+            self.log.error(
+                "No Ansible Runner Service address. "
+                "Try 'ceph config set mgr mgr/%s/server_addr <server name/ip>'",
+                self.module_name)
+
+        if not self.get_config('server_port', ''):
+            self.log.error(
+                "No Ansible Runner Service port. "
+                "Try 'ceph config set mgr mgr/%s/server_port <port number>'",
+                self.module_name)
+
+        if not self.get_config('username', ''):
+            self.log.error(
+                "No Ansible Runner Service user. "
+                "Try 'ceph config set mgr mgr/%s/username <string value>'",
+                self.module_name)
+
+        if not self.get_config('password', ''):
+            self.log.error(
+                "No Ansible Runner Service User password. "
+                "Try 'ceph config set mgr mgr/%s/password <string value>'",
+                self.module_name)
+
+# Auxiliary functions
+#==============================================================================
+
+def process_inventary_json(inventory_events, ar_client, playbook_uuid):
+    """ Adapt the output of the playbook used in 'get_inventory'
+        to the Orchestrator expected output (list of InventoryNode)
+
+    @param inventory_events: events dict with the results
+
+        Example:
+        inventory_events =
+        {'37-100564f1-9fed-48c2-bd62-4ae8636dfcdb': {'host': '192.168.121.254',
+                                                    'task': 'RESULTS',
+                                                    'event': 'runner_on_ok'},
+        '36-2016b900-e38f-7dcd-a2e7-00000000000e': {'host': '192.168.121.252'
+                                                    'task': 'RESULTS',
+                                                    'event': 'runner_on_ok'}}
+    @param ar_client: Ansible Runner Service client
+    @param playbook_uuid: Palybooud identifier
+
+    @return              : list of InventoryNode
+    """
+
+    #Obtain the needed data for each result event
+    inventory_nodes = []
+
+    # Loop over the result events and request the event data
+    for event_key, data in inventory_events.iteritems():
```

Comment:
`iteritems` is not supported in Python 3

Reply:
Removed!. Thanks!